java-squid/src/main/java/org/sonar/java/se/symbolicvalues/BinaryRelation.java

```diff
@@ -35,6 +35,49 @@
 
 public abstract class BinaryRelation {
 
+
+  public static BinaryRelation binaryRelation(Kind kind, SymbolicValue leftOp, SymbolicValue rightOp) {
+    BinaryRelation relation;
+    switch (kind) {
+      case EQUAL:
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Reduce this switch case number of lines from 8 to at most 5, for example by extracting code into methods. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1151)